typing/ctype.ml

```diff
@@ -3160,9 +3166,9 @@ let unify env ty1 ty2 =
   try
     unify env ty1 ty2
   with
-    Unify trace ->
+    Unify_trace trace ->
       undo_compress snap;
-      raise (Unify (expand_trace !env trace))
+      raise (Unify {trace = expand_trace !env trace})
```

Comment:
Could it be worthwhile to move the finalization of the trace into a complete error to `Errortrace`?

Reply:
We can't move `expand_trace` into `Errortrace`, as it depends on the expansion machinery defined in `Ctype`.  Changing the types, as you suggested above, I think produces most of the same benefit you were looking for here.